content/en/platform/corda/4.8/enterprise/node/deploy/generating-a-node-dockerform.md

```diff
@@ -0,0 +1,346 @@
+        ALTER DEFAULT privileges IN SCHEMA \$dbSchema GRANT SELECT, INSERT, UPDATE, DELETE, REFERENCES ON tables TO \$dbUser;
+        GRANT USAGE, SELECT ON ALL sequences IN SCHEMA \$dbSchema TO \$dbUser;
+        ALTER DEFAULT privileges IN SCHEMA \$dbSchema GRANT USAGE, SELECT ON sequences TO \$dbUser;
+        ALTER ROLE \$dbUser SET search_path = \$dbSchema;
+EOSQL
+""".replaceAll("\r\n", "\n")
+        }
+        initialScript.executable = true
+    }
+}
+```
+
+2. In the `build.gradle` file, add the gradle task `generateInitScripts` to the `dependsOn` list of the `prepareDockerNodes` task, add the `dockerConfig` element, and initialise it with the `postgres` block. An example is shown below:
+
+```groovy
+task prepareDockerNodes(type: net.corda.plugins.Dockerform, dependsOn: ['jar',  'generateInitScripts']) {
+
+    [...]
+
+    node {
+        [...]
+    }
+
+    // The postgres block from the postgres.gradle file
+    dockerConfig = postgres
+}
+```
+
+The `postgres.gradle` file includes the following:
+* A gradle task called `generateInitScripts` used to generate the Postgres Docker image files.
+* A set of variables used to initialise the Postgres Docker image.
+
+To set up the external database, you must place the following two files in the `build` directory:
+* `Postgres_Dockerfile` -  a wrapper for the base Postgres Docker image.
+* `Postgres_init.sh` -  a shell script to initialise the database.
+
+The `Postgres_Dockerfile` is referenced in the `docker-compose.yml` file and allows for a number of arguments for configuring the Docker image.
+
+You can use the following configuration parameters in the `postgres.gradle` file:
+
+| Parameter              | Description                                       |
+|------------------------|---------------------------------------------------|
+| postgresql_version     | Version of JDBC driver to connect to the database |
+| postgres_image_version | Version of Postgres Docker image                  |
+| dbUser                 | Database user                                     |
+| dbPassword             | Database password                                 |
+| dbSchema               | Postgres schema                                   |
+| dbName                 | Database name                                     |
+| dbPort                 | Database port (default: 5432)                     |
+| dbHostName             | Database host (default: localhost)                |
+| dbInit                 | Initialisation script for Postgres Docker image   |
+| dbDockerfile           | Wrapper of base Postgres Docker image             |
+| dbDataVolume           | Path to database files for Postgres Docker image  |
+
+To make the database files persistent across multiple `docker-compose` runs, you must set the `dbDataVolume` parameter. If this variable is commented out, the database files will be removed after every `docker-compose` run.
+
+#### Run the Dockerform task
```

Comment:
These header values are a bit off. We should really only be using H4 for very detailed information. Can you take a look at rationalising these a bit.